layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -319,6 +319,14 @@ initialized with the current filename."
                (new-name (read-file-name "New name: " (if arg dir filename))))
           (cond ((get-buffer new-name)
                  (error "A buffer named '%s' already exists!" new-name))
+                ((string-equal new-name filename)
+                 (message
+                  (propertize
+                   (format "File renaming failed! Same new and old name: '%s'"
+                           name)
+                   'face 'warning))
+                 (sit-for 2)
```

Comment:
Doesn't look optimal to me as well. But it works.

I see the described problem with `ivy` as well.